src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -0,0 +1,75 @@
+using System.Text;
+using System.Web;
+using System.Xml;
+using Amazon.SecurityToken;
+using Amazon.SecurityToken.Model;
+
+namespace D2L.Bmx.Aws;
+
+public interface ICloudProvider<TRoleState> where TRoleState : IRoleState {
+	TRoleState GetRoles( string encodedSaml );
+	Task<Dictionary<string, string>> GetTokensAsync( TRoleState state, int selectedRoleIndex );
+}
+public class AwsClient : ICloudProvider<AwsRoleState> {
+	private readonly IAmazonSecurityTokenService _stsClient;
+
+	public AwsClient( IAmazonSecurityTokenService stsClient ) {
+		_stsClient = stsClient;
+	}
+
+	private XmlDocument ParseSamlToken( string encodedSaml ) {
+		var samlStatements = encodedSaml.Split( ";" );
+		// Process the B64 Encoded SAML string to get valid XML doc
+		var samlString = new StringBuilder();
+		foreach( var inputValueString in samlStatements ) {
+			samlString.Append( HttpUtility.HtmlDecode( inputValueString ) );
```

Comment:
`HttpUtility` is from the `System.Web` namespace, which is... legacy, and is even shitted on the .NET team themselves.

We can use [WebUtility](https://learn.microsoft.com/en-us/dotnet/api/system.net.webutility?view=net-7.0) as an alternative.